README.md

```diff
@@ -1,9 +1,236 @@
 # CDK Framework
 
-The CDK Framework helps CDK authors build CDKs.
+The CDK Framework helps CDK authors build CDKs for the [Internet Computer](https://internetcomputer.org/).
 
 ## Core Concepts
 
 Start with the AST of your language and compile it into an Abstract Canister Tree (a concept from the CDK Framework). The CDK Framework will then create a Rust canister which can be compiled into Wasm and deployed to the IC.
 
 AST -> ACT -> Rust canister code -> Wasm binary -> IC
+
+## Disclaimer
+
+Please consider the [security section](#security).
+
+## Contents
+
+- [Usage](#usage)
+  - [Installation](#installation)
+  - [Abstract Canister Tree](#abstract-canister-tree)
+  - [Candid Types](#candid-types)
+  - [Canister Methods](#canister-methods)
+  - [External Canisters](#external-canisters)
+  - [Guard Functions](#guard-functions)
+  - [Rust Parts](#rust-parts)
+- [Advanced Usage](#advanced-usage)
+  - [Inline Names](#inline-names)
+  - [Flatten](#flatten)
+  - [ToTypeAnnotation](#to-type-annotation)
+  - [Traits](#traits)
+- [Security](#security)
+- [Contributing](#contributing)
+
+## Usage
+
+### Installation
+
+You should be using a \*nix environment (Linux, Mac OS, [WSL](https://learn.microsoft.com/en-us/windows/wsl/install)) with bash and have Rust and dfx installed on your system.
+
+#### dfx
+
+Run the following command to install dfx 0.12.1:
+
+```bash
+DFX_VERSION=0.12.1 sh -ci "$(curl -fsSL https://sdk.dfinity.org/install.sh)"
+```
+
+#### Your CDK
```

Comment:
This section is good, but doesn't belong in the installation section. It should be in a separate place. Like, they need to make an ACT, and generate a Cargo.toml, etc. That's where this belongs.